### ISSUE TITLE
Installing on Arch Linux

### ISSUE THREAD

**Issue by madskjeldgaard** [leomccormack/SPARTA#20]
Hi, sorry I know there's an issue open on linux installation right now, but I thought I'd make a seperate one for installing on Arch Linux (and once this is succesful I will make a package for the aur). 

I managed to install most/all dependencies via yay:

```
yay -S hdf5 libxext lapack libxinerama webkitgtk2-bin netcdf-fortran libcurl-gnutls
```
The only problem now is that it is still complaining about a missing cdff file:
```
swell: dlopen() failed: libnetcdf.so.13: cannot open shared object file: No such file or directory
```
libnetcdf.so.13 is not included in any of the netcdf-fortran packages on Arch as far as I can tell. Any pointers?

**Comment by madskjeldgaard** [comment]
This symlink solved the libnetcdf.so.13 issue and made the whole thing work properly it seems :)

```
sudo ln -s /usr/lib/libnetcdf.so /usr/lib/libnetcdf.so.13
```

**Comment by madskjeldgaard** [comment]
Okay, I have most of the package done now but a few issues remain. Only some of the plugins seem to work. `compass_binaural` and `compass_upmixer` seem to work fine for example, but eg `compass_decoder` makes Reaper crash without any errors in either terminal or `journalctl -f`. Any idea what the underlying problem is here since some work and some don't ?

**Comment by leomccormack** [comment]
The compass plugins are undergoing a bit of an overhaul at the moment, which may fix this problem; (although, also perhaps not as I don't recall such a problem). However, please feel to re-open this issue if compass_decoder still gives you problems after the next release : )

On a side-note, if you are willing to put instructions specifically for Arch linux down on paper (.md) and make a PR. It may help others in future (perhaps adding to: https://github.com/leomccormack/Spatial_Audio_Framework/blob/develop/dependencies/SOFA_READER_MODULE_DEPENDENCIES.md ?). 
No pressure of course : )

**Comment by madskjeldgaard** [comment]
Thanks ! I will... i have an arch package for these plugins ready to go once the linux build works a bit better and then it's easy breasy. Let me know if you want me to test anything :-) 




On Tue, Sep 29, 2020 at 5:20 PM +0200, "Leo McCormack" <notifications@github.com> wrote:















Closed #20.

**Comment by madskjeldgaard** [comment]
The problem with libnetcdf persist with the new (otherwise very nice release!!!) FYI. E.g. Ambi bin won't load, unless you symbolically link:
`sudo ln -s /usr/lib/libnetcdf.so.18 /usr/lib/libnetcdf.so.15`

**Comment by madskjeldgaard** [comment]
> The problem with libnetcdf persist with the new (otherwise very nice release!!!) FYI. E.g. Ambi bin won't load, unless you symbolically link:
> `sudo ln -s /usr/lib/libnetcdf.so.18 /usr/lib/libnetcdf.so.15`

And even then it seems to crash randomly

**Comment by madskjeldgaard** [comment]
I figured out the final issues.
The errors pop up because the prebuilt linux release is dependent on old versions of netcdf and hdf5 on your system.
This can be solved on arch by downgrading
`downgrade hdf5`
set to 1.10.5 (for now)
and
`downgrade netcdf`
to 4.7.3

**Comment by madskjeldgaard** [comment]
There is a ready to use PKGBUILD file here for installing the Sparta plugins and the comments have the information about downgrading. I will consider putting it on the AUR later if I can make it more stable with the dependencies.

https://gist.github.com/madskjeldgaard/21743f62cbb6f38b6e83c18cffe67ae3

**Comment by madskjeldgaard** [comment]
Hi again @leomccormack  

I've now added the plugins to the Arch User Repository and users of arch based ssystems may now install them using an AUR helper like so: `paru -S sparta-plugins`.

The only thing missing is the binaural plugins - I cannot get them to show up on my system even though they are installed. I've installed all dependencies mentioned in the install instructions (I think): `"hdf5" "netcdf" "netcdf-fortran" "libglvnd" "freetype2" "libxinerama" "webkit2gtk" "gtk3" "libxext" "lapack" "libcurl-gnutls"`. Any other pointers?
